api/clamav_scanner/scan.py

```diff
@@ -0,0 +1,84 @@
+import datetime
+import json
+import os
+
+from .common import AV_DEFINITION_S3_BUCKET
+from .common import AV_DEFINITION_S3_PREFIX
+from .common import AV_SIGNATURE_METADATA
+from .common import AV_STATUS_METADATA
+from .common import AV_TIMESTAMP_METADATA
+from .common import AWS_ENDPOINT_URL
+
+from boto3wrapper.wrapper import get_session
+from clamav_scanner.clamav import determine_verdict, update_defs_from_s3, scan_file
+from database.db import get_db_session
+from logger import log
+from models.Scan import Scan, ScanProviders
+
+
+def sns_scan_results(sns_client, scan, sns_arn, scan_signature):
+
+    message = {
+        "scan_id": scan.id,
+        AV_SIGNATURE_METADATA: scan_signature,
+        AV_STATUS_METADATA: scan.verdict,
+        AV_TIMESTAMP_METADATA: scan.completed,
+    }
+
+    sns_client.publish(
+        TargetArn=sns_arn,
+        Message=json.dumps({"default": json.dumps(message, default=str)}),
+        MessageStructure="json",
+        MessageAttributes={
```

Comment:
When we add the endpoint that has Scan Files download the file from S3 itself, can we add the `Bucket` and `Object Key` to the `MessageAttributes?  That'll make it dead simple to update the target object with the scan status.

Alternatively we could tag the S3 object with the `scan_id` and then when the SNS message comes back, filter the S3 objects by that tag value to find the target object.

Reply:
i'm with you on that one. better to send the object key so that we don't have to search all s3 objects. I've made a note of it when adding the s3 direct read endpoint